idaes/models/properties/modular_properties/transport_properties/tests/test_no_method.py

```diff
@@ -0,0 +1,132 @@
+#################################################################################
+# The Institute for the Design of Advanced Energy Systems Integrated Platform
+# Framework (IDAES IP) was produced under the DOE Institute for the
+# Design of Advanced Energy Systems (IDAES), and is copyright (c) 2018-2021
+# by the software owners: The Regents of the University of California, through
+# Lawrence Berkeley National Laboratory,  National Technology & Engineering
+# Solutions of Sandia, LLC, Carnegie Mellon University, West Virginia University
+# Research Corporation, et al.  All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and
+# license information.
+#################################################################################
+"""
+Test no_method method to specify that no method has been assigned to calculate
+a physical property for a phase.
+
+Authors: Douglas Allan
+"""
+
+import pytest
+from sys import modules
+
+from pyomo.environ import ConcreteModel, units as pyunits
+
+from idaes.models.properties.modular_properties.transport_properties import NoMethod
+
+from idaes.core import declare_process_block_class, LiquidPhase, VaporPhase
+
+from idaes.models.properties.modular_properties.base.generic_property import (
+    GenericParameterData,
+)
+from idaes.models.properties.modular_properties.eos.ideal import Ideal
+from idaes.models.properties.modular_properties.pure import ConstantProperties
+
+
+@declare_process_block_class("DummyParameterBlock")
+class DummyParameterData(GenericParameterData):
+    def configure(self):
+        self.configured = True
+
+    def parameters(self):
+        self.parameters_set = True
+
+
+def define_state(b):
+    b.state_defined = True
+
+
+# Dummy method to avoid errors when setting metadata dict
+def set_metadata(b):
+    pass
+
+
+def construct_dummy_model(component_dict):
```

Comment:
Does this actually get used in these tests or can it be removed?

Reply:
We do not need a real state block to log the error message, but we do need a parameter block. The most I'd want to do is eliminate the `component_dict` argument, I think

Reply:
Ah - I missed the use of this method in the tests.